core/test/nuxeo-audit-page-provider.test.js

```diff
@@ -0,0 +1,212 @@
+const to = moment()
+  .add(5, 'days')
+  .format('YYYY-MM-DD');
+const requestParameters = {
+  startEventDate: from,
+  endEventDate: to,
+  category: eventDocumentCategory,
+  eventId: events,
+  principalName: user,
+};
+const providerParameters = {
+  startDate: from,
+  endDate: to,
+  eventCategory: eventDocumentCategory,
+  eventIds: events,
+  principalName: user,
+};
+const providerSort = {
+  eventDate: 'asc',
+  category: 'desc',
+  eventId: 'desc',
+  principalName: 'asc',
+};
+let server;
+async function getProvider(params = {}, docId = documentId, sort = {}) {
+  const provider = await fixture(
+    html`
+      <nuxeo-audit-page-provider
+        id="provider"
+        doc-Id="${docId}"
+        params="${JSON.stringify(params)}"
+        sort="${JSON.stringify(sort)}"
+        page-size="40"
+      ></nuxeo-audit-page-provider>
+    `,
+  );
+  return provider;
+}
+
+function mockDocHistoryResponse(qParams = {}) {
+  // build the expected parameters (add the commons properties)
+  const expectedQueryParam = { currentPageIndex: 0, pageSize: 40, ...qParams };
+  server.respondWith('GET', `/api/v1/id/${documentId}/@audit`, ({ queryParams }) => {
+    expect(expectedQueryParam).to.eql(queryParams);
+    return { 'entity-type': 'logEntries', entries: [] };
+  });
+}
+
+function mockAuditResponse(requestParams = {}) {
+  server.respondWith('POST', '/api/v1/automation/Audit.QueryWithPageProvider', ({ body }) => {
+    // build request parameters and sort (optional) from the body
+    const { currentPageIndex, pageSize, providerName, namedQueryParams, sortBy, sortOrder } = body.params;
+    const flatedQueryParams = {};
+    Object.entries(namedQueryParams).forEach(([key, value]) => {
+      flatedQueryParams[key] = key === 'eventIds' ? JSON.parse(value) : value;
+    });
+    let bodyParams = { currentPageIndex, pageSize, providerName, ...flatedQueryParams };
+    if (sortBy && sortOrder) {
+      bodyParams = { sortBy, sortOrder, ...bodyParams };
+    }
+
+    // build the expected parameters (add the commons properties)
+    const expectedRequestParam = { currentPageIndex: 0, pageSize: 40, providerName: 'EVENTS_VIEW', ...requestParams };
+
+    expect(expectedRequestParam).to.eql(bodyParams);
+    return { 'entity-type': 'logEntries', entries: [] };
+  });
+}
+
+function checkResponse(response) {
+  expect(response).to.be.an('object');
+  expect(response).to.have.property('entity-type');
+  expect(response).to.have.property('entries');
+  expect(response.entries).to.be.an('array').that.is.empty;
+}
+
+suite('nuxeo-audit-page-provider', () => {
+  setup(() => {
+    server = fakeServer.create();
+  });
+
+  teardown(() => {
+    server.restore();
+  });
+
+  suite('Fetch Document History', () => {
+    test('Should get history without parameters', async () => {
+      mockDocHistoryResponse();
+      const provider = await getProvider({}, documentId);
+      checkResponse(await provider.fetch());
```

Comment:
Following the comment above, if we want to perform some assertions to the request, we can get it from the `nuxeo-mock-client` with `getLastRequest` or `getRequests` methods ([here](https://github.com/nuxeo/nuxeo-elements/blob/master/testing-helpers/nuxeo-mock-client.js#L103)).

Then, we can perform all kind of assertions to the request sent out 🙂 ([example](https://github.com/nuxeo/nuxeo-elements/blob/master/ui/test/nuxeo-document-comment.test.js#L331-L339))

Reply:
👍